src/Interpreters/FullSortingMergeJoin.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <Interpreters/IJoin.h>
+#include <Interpreters/TableJoin.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <Poco/Logger.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int TYPE_MISMATCH;
+    extern const int NOT_IMPLEMENTED;
+}
+
+/// Dummy class, actual joining is done by MergeTransform
+class FullSortingMergeJoin : public IJoin
+{
+public:
+    explicit FullSortingMergeJoin(std::shared_ptr<TableJoin> table_join_, const Block & right_sample_block_)
+        : table_join(table_join_)
+        , right_sample_block(right_sample_block_)
+    {
+        LOG_TRACE(&Poco::Logger::get("FullSortingMergeJoin"), "Will use full sorting merge join");
+    }
+
+    const TableJoin & getTableJoin() const override { return *table_join; }
+
+    bool addJoinedBlock(const Block & /* block */, bool /* check_limits */) override { __builtin_unreachable(); }
```

Comment:
Usually we throw NOT_IMPLEMENTED for virtual functions.
Maybe __builtin_unreachable is also ok, but a little bit unusual

Reply:
yes, LOGICAL_ERROR is better